core/sql/exp/exp_conv.cpp

```diff
@@ -9321,6 +9321,32 @@ convDoIt(char * source,
   };
   break;
 
+// gb2312 -> utf8
+  case CONV_GBK_F_UTF8_V:
+  {
+    char * targetbuf = new char[sourceLen*4+1];
+    size_t sl = sourceLen;
+    int convLen = gbkToUtf8( source, sl, targetbuf, sl*4);
+    int copyLen = 0;
+    if (convLen > 0) {
+      copyLen = (convLen< targetLen) ? convLen: targetLen;
```

Comment:
What if there is a real overflow, i.e. user has given a GBK value which when converted to UTF8 exceeds the max varchar length? Will that be covered by this error or is that checked in a different place? Some how I don't see max length of target column in here...